docs/design/resource-states.md

```diff
@@ -0,0 +1,222 @@
+| WaitingForOwner        | Warning                         | The `owner` of this resource cannot be found in Kubernetes. Progress is blocked until the `owner` is created.                                                             |
+| Deleting               | Info                            | The resource is being deleted.                                                                                                                                            |
+| Succeeded              | None (""/empty)                 | The `spec` has successfully been applied. No additional changes are being attempted at this time.                                                                         |
+
+There are no failure conditions (`Severity = Error`) specified here because currently all fatal errors come directly from Azure. When an error response is received from Azure, the `Code` from Azure is set as the `Reason`, and the `Message` from Azure
+is set as the `Message`.
+
+#### `Severity` meaning in the context of the `Ready` condition
+
+* `Info` means that everything is proceeded as expected. This is the "happy path".
+* `Warning` means that something is wrong, but we haven't given up. The resource will continue to be reconciled until we can progress past the cause of the `Warning`. Users should examine the `Warning` as some may be due to the operator waiting for action from them, as in the case of `Reason=WaitingForOwner, Severity=Warning`. Others may be due to transient unavailability in Azure, as in the case of `Reason=InternalServerError, Severity=Warning`.
+* `Error` means that we were unable to reconcile the resource successfully. A `Ready` condition with `Status=False, Reason=<something>, Severity=Error` is no longer attempting to be reconciled. The user must make an update to the resource to fix one or more configuration problems.
+
+
+## Reference
+1. [Article on conditions and status reporting in Kubernetes](https://dev.to/maelvls/what-the-heck-are-kubernetes-conditions-for-4je7). Note that this article is a bit old as Cluster API uses conditions now, but it gives a great overview of the topic.
+2. [Reccomended shape of Conditions](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1623-standardize-conditions)
+3. [Update condition guidance](https://github.com/kubernetes/community/pull/4521) - and the actual guidance itself is [here](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties)
+4. [More condition reccomendations](https://github.com/kubernetes-sigs/cluster-api/issues/1658#issuecomment-584794287)
+
+## Open questions
+1. There are some small differences between this proposal and what CAPI is doing
+   * CAPI says `Reason` is optional, but we're making it required.
+   * CAPI's `Type` field is an alias to `string` called `ConditionType` and ours is just `string`.
```

Comment:
I've changed my mind on this one -- will update to use a string alias and define a set of possible values. That more clearly shows that what we currently support is just `Ready` and if we add more they should be added uniformly across the operator (we don't have the same sub-project situation that CAPI does)